lib/services/credentials/credential.dao.js

```diff
@@ -114,6 +114,7 @@ dao.insertCredential = function (id, type, credentialObj) {
   }
   const key = buildIdKey(type, id);
   if (type === 'key-auth') {
+    if (credentialObj.keyId) credentialObj.id = credentialObj.keyId;
```

Comment:
it looks suspicious to me. isn't keyId some separate property? 
there is `id` incoming. As I understand it is used to reference key-auth credentials. so it could break listing creds 

The related task is about password hash, why there are changes to key-auth at all ?

Reply:
So the problem here is that — we have yet another inconsistency when returning the credentials. Some of them return an `id` as an identifier, some of them return `keyId`.

So when you use the API you don't know where the id is stored. You need to look into `id` and if it does not exist, you look for `keyId` — this is confusing (it was confusing an user).

I agree that's suspicious *and* an horrible solution, but I haven't found anything better without breaking the API. Open to better solutions!